keps/sig-apps/2255-pod-cost/README.md

```diff
@@ -55,102 +48,130 @@ Items marked with (R) are required *prior to targeting to a milestone / release*
+This feature allows applications to give a hint to the ReplicaSet controller 
+as to which pods should be deleted first on scale down.
 
 ## Motivation
 
-For some applications, it is necessary that the application can tell Kubernetes which pod can be deleted and which replica has to be protected. The reason for this is that some applications do have stateful sessions and it is not possible to put such an application into Kubernetes because of session termination resulting from "random" down-scale. If the application is able to tell Kubernetes which of the replicas contains no/few/less important active sessions, this would solve many problems. This feature is non-disruptive to the default behaviour. Only if the annotation is existing, it will make a difference in deletion order.
+Currently ReplicaSets are scaled down based on a criteria that on the
+limit prioritizes deleting pods with a more recent creation/readiness 
+timestamp. This is not ideal for some applications where the cost of 
+deleting pods is not related to how recent they were created. 
 
 ### Goals
 
-To recommend which pod gets deleted next of a ReplicaSet. This should help to avoid major reworks in existing applications architecture:
-* [45509](https://github.com/kubernetes/kubernetes/issues/45509) - Scale down a deployment by removing specific pods
+- Allow applications to influence the order of deleting pods when scaling down a ReplicaSet
 
 
 ### Non-Goals
 
-Guaranteed (in contrast to the recommendation stated in Goals) deletion of a selected replica.
+- Guarantees on pod deletion order
 
 ## Proposal
 
-The application can set the `controller.kubernetes.io/pod-cost` annotation to a pod through the Kubernetes API. When a downscale event happens, the pod with the lower priority value of the previously set annotation will be deleted first. If one pod of the Deployment has no priority annotation set, it will be treated as the lowest priority.
+Define a known annotation, namely `controller.kubernetes.io/pod-cost` that 
+applications can set to offer a hint on the cost of deleting a pod compared
+to other pods belonging to the same ReplicaSet. 
 
-If all pods have the same priority, there is no difference in the normal pod delete decision behaviour. The same applies if the pod-cost annotation is not used at all.
+The pod-cost range will be from [-MaxInt, MaxInt]. The default value is 0.
+Invalid values (like setting the annotation to string) will be converted to
+the default value.
 
-The pod-cost annotation can be changed during operation, for example, if workload changes or a new master gets elected.
+When scaling down a ReplicaSet, controller-manager will prioritize deleting
+pods with lower pod-cost. Specifically, the rank passed to 
+[ActivePodsWithRanks](https://github.com/kubernetes/kubernetes/blob/cac933934b1301665e6e51a81c66c483f4e16c49/pkg/controller/controller_utils.go#L784-L809)
+is pod-cost inverse then number of pods on node (the latter is the current 
```

Comment:
We don't need to go into implementation details, but I find this sentence rather confusing.

There is only one rank number (step 4 in the code comment), so I don't know how to interpret the word "then"

Reply:
let me try to rephrase.

Reply:
ok, done, how about now?

Reply:
lgtm